tests/platforms/android/apk/test_verify_tools.py

```diff
@@ -0,0 +1,183 @@
+import subprocess
+from io import BytesIO
+from stat import S_IMODE
+from unittest import mock
+from zipfile import ZipFile
+
+import pytest
+from requests import exceptions as requests_exceptions
+
+from briefcase.exceptions import BriefcaseCommandError, NetworkFailure
+from briefcase.platforms.android.apk import ApkBuildCommand
+
+
+def create_sentinel_zipfile():
+    out = BytesIO()
+    with ZipFile(out, "w") as zipfile:
+        zipfile.writestr("tools/bin/sdkmanager", "")
+    return out.getvalue()
+
+
+@pytest.fixture
+def build_command(tmp_path, first_app_config):
+    class TestableApkBuildCommand(ApkBuildCommand):
+        dot_briefcase = tmp_path / ".briefcase" / "tools"
+
+    command = TestableApkBuildCommand(
+        base_path=tmp_path / "base_path", apps={"first": first_app_config},
+    )
+    command.tmp_path = tmp_path
+    command.host_os = "Linux"
+    command.os = mock.MagicMock()
+    command.os.environ = {}
+    command.sys = mock.MagicMock()
+    command.sys.version_info.major = 3
+    command.sys.version_info.minor = 7
+    command.requests = mock.MagicMock()
+    command.subprocess = mock.MagicMock()
+    return command
+
+
+def test_sdk_url(build_command):
+    build_command.host_os = "Linux"
+    assert build_command.sdk_url == (
+        "https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip"
+    )
+
+
+def test_permit_python_37(build_command):
+    build_command.verify_python_version()
```

Comment:
A comment would be helpful here; it wasn't immediately obvious to me why this would pass under Python != 3.7.

Reply:
+1

Reply:
Resolved in latest push